redux/main.py

```diff
@@ -91,11 +91,12 @@ def __init__(
         self._event_handlers_queue = queue.Queue[
             tuple[EventHandler[Event], Event] | None
         ]()
-        self_workers = [
+        self._workers = [
             _SideEffectRunnerThread(self._event_handlers_queue)
             for _ in range(self.store_options.threads)
         ]
-        for worker in self_workers:
+        for worker in self._workers:
+            worker.setDaemon(True)
```

Comment:
Have you run the demos? They seem to quit without any issues.
We should keep the threads as non-daemon because this helps to make sure we are quitting the threads properly, if we switch to daemon threads without finding out why they are not closing properly, we might overlook a fundamental design problem by silencing it.
Could you create an example where the threads fail to quit? This will give us a solid basis for our discussion.

Reply:
Yes, I agree with you, lets keep the threads as non-daemon.
I tried several times, it only happens when hit error in python interactive mode.